### ISSUE TITLE
Filtering Search Results - "Snapshot" isn't cleared

### ISSUE THREAD

**Issue by grahamsinger** [tsgrp/HPI#1251]
In this example, a search is done on two different types.  When attempting to filter results on the second type, we get an error because the previous "snapshot":
```  
//before we filter - create a snapshot of the current state to go back to
snapshotState: function() {
       this.snapshot = new OCQuery._Snapshot(_.extend({}, this.records.attributes), this.fullCollection);  
},
```
is not reset.  This causes the [findWhere](http://backbonejs.org/#Collection-findWhere) to fail because the `currentSearchConfig()` is updates, but the snapshot is not.

See [here](http://g.recordit.co/bZnDVypNAO.gif) (sorry it's a separate link).

Related to #1249

**Comment by gsteimer** [comment]
@grahamsinger - if I'm reading this right, this means that once you filter on a certain type, filters on other types won't work until you refresh the page.  Is that correct?

If the above is true (or even half true), I think we need to fix this soon, so I'm bumping the priority to 2.5.

**Comment by grahamsinger** [comment]
@gsteimer yeah that's correct.  I'm about to finish #525, and then I'll look into #1249 and this.

**Comment by peterkk** [comment]
Fixed with HPI commit #16766

**Comment by grahamsinger** [comment]
Reverted fix for this.  IMO, this needs to be branched and coordinated if multiple are going to work on it.  I've seen a handful of instances where "fixing" something results in breaking something else.

**Comment by peterkk** [comment]
Closed with HPI commit #16783
